projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -4,222 +4,263 @@ import apiFetch from '@wordpress/api-fetch';
+				__( "This is a post titled '%1$s', published in categories '%2$s':\n", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+				title,
+				categoryNames
+			);
+		}
+
+		// Title and content
+		if ( content ) {
+			return sprintf(
+				/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and the last 240 characters of content. */
+				__( "This is a post titled '%1$s':\n\n…%2$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+				title,
+				content
+			);
+		}
+
+		return sprintf(
+			/** translators: This will be a prompt to OpenAI to generate a post based on the post title */
+			__( 'Write content of a post titled "%s"', 'jetpack' ),
+			title
+		);
+	};
+
+	const containsAiUntriggeredParapgraph = _content =>
+		_content.filter(
+			block => block.name && block.name === 'jetpack/ai-paragraph' && ! block.attributes.content
+		).length > 0;
+
+	const taxonomies = categoryObjects.filter( cat => cat.id !== 1 ).concat( tagObjects );
 	const categoryNames = taxonomies.map( ( { name } ) => name ).join( ', ' );
+	const contentBefore = useSelect( select => {
+		const editor = select( 'core/block-editor' );
+		const index = editor.getBlockIndex( clientId );
+		return editor.getBlocks().slice( 0, index ) ?? [];
+	} );
 
-	return createPrompt( currentPost.title, editorContent, categoryNames );
-}
+	const content = contentBefore
+		.filter( function ( block ) {
+			return block && block.attributes && block.attributes.content;
+		} )
+		.map( function ( block ) {
+			return block.attributes.content.replaceAll( '<br>', '\n' );
+		} )
+		.join( '\n' );
+	const contentToUseForPrompt = createPrompt( currentPostTitle, content, categoryNames );
+
+	const getSuggestionFromOpenAI = () => {
+		if ( completionFinished || loadingCompletion ) {
+			return;
+		}
 
-export default function Edit( { attributes, setAttributes } ) {
-	const [ loadingCompletion, setLoadingCompletion ] = useState( true );
-	const [ errorMessage, setErrorMessage ] = useState( '' );
-
-	// Here is where we craft the prompt.
-	const formattedPrompt = useSelect( preparePromptBasedOnEditorState, [] );
-
-	//useEffect hook is called only once when block is first rendered.
-	useEffect( () => {
-		//Theoretically useEffect would ensure we only fire this once, but I don't want to fire it when we get data to edit either.
-		if ( ! attributes.content && ! attributes.requestedPrompt ) {
-			getSuggestionFromOpenAI(
-				setAttributes,
-				formattedPrompt,
-				setLoadingCompletion,
-				setErrorMessage
+		setShowRetry( false );
+		setErrorMessage( false );
+		setNeedsMoreCharacters( 0 );
+		setLoadingCompletion( true );
+
+		const data = { content: contentToUseForPrompt };
+		apiFetch( {
+			path: '/wpcom/v2/jetpack-ai/completions',
+			method: 'POST',
+			data: data,
+		} )
+			.then( res => {
+				const result = res.prompts[ 0 ].text;
+
+				// We set it up so it doesn't start with nothing
```

Comment:
This comment seems incorrect (it sound to me like something that would get set before the `apiFetch` was called instead of after the api call has occurred).

Reply:
It was actually added at some point where I had to display the first word of the answer otherwise a some `attributes.content` check was messing stuff up. I'll remove. Nice catch!

Reply:
Done in a6869f84afce9f4647198e86d73381ee7a80faa3